test/common/validate_resource_limits.go

```diff
@@ -0,0 +1,102 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"github.com/integr8ly/integreatly-operator/apis/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/pkg/resources/quota"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+)
+
+var (
+	productsToTest = map[v1alpha1.ProductName]map[string]string{
+		v1alpha1.Product3Scale: {
```

Comment:
@makslion are we planning to introduce testing the sidecar containers at a later stage?

Reply:
we could add more containers to test resources in the future. Currently, we are [testing](https://github.com/integr8ly/integreatly-operator/pull/2635/files#diff-1bb321a94b2c8abb72a7b0f876f890db05e14a1223fac5b40d11e7769a149075R96) sidecars (if they are present) in all of the mentioned pods. To be more accurate - this test checks all the containers and ensures that limits and requests are set to the expected value for the "main" container and that they are present in all other containers

Reply:
Please hold with merging this PR, let's chat about it :)